src/mumble/MainWindow.cpp

```diff
@@ -534,12 +539,16 @@ void MainWindow::closeEvent(QCloseEvent *e) {
 			return;
 		}
 
-		if (qcbRemember->isChecked()) {
-			Global::get().s.connectedQuitBehavior = ConnectedQuitBehavior::CLOSE;
+		/* If checkbox is checked and not connected, keep asking when connected
+		 * If checkbox is checked and connected, quit always */
+		if (qcbRemember->isChecked() && !(sh && sh->isRunning())) {
+			Global::get().s.quitBehavior = QuitBehavior::ASK_WHEN_CONNECTED;
+		} else if (qcbRemember->isChecked() && sh && sh->isRunning()) {
+			Global::get().s.quitBehavior = QuitBehavior::ALWAYS_QUIT;
 		}
-	} else
 #endif
-		if (sh && sh->isRunning() && Global::get().s.connectedQuitBehavior == ConnectedQuitBehavior::MINIMIZE) {
+	} else if (!bSuppressAskOnQuit && ((quitBehavior == QuitBehavior::ALWAYS_MINIMIZE)
+		   || (sh && sh->isRunning() && quitBehavior == QuitBehavior::MINIMIZE_WHEN_CONNECTED))) {
```

Comment:
TODO: Factor bool expressions out in order to give them names (as is done above)